beacon-chain/powchain/service.go

```diff
@@ -284,6 +288,32 @@ func (w *Web3Service) handleDelayTicker() {
 		log.Error(err)
 	}
 }
+func (w *Web3Service) rerunRPCClientAndSubscribe(headSub ethereum.Subscription, header *gethTypes.Header) {
+	rpcClient, err := gethRPC.Dial(w.endpoint)
+	if err != nil {
+		log.Errorf("Access to PoW chain is required for validator. Unable to connect to Geth node: %v", err)
+	}
+	headSub.Unsubscribe()
+	powClient := ethclient.NewClient(rpcClient)
+	w.client = powClient
+	w.reader = powClient
+	w.blockFetcher = powClient
+	w.logger = powClient
+	headSub, err = w.reader.SubscribeNewHead(w.ctx, w.headerChan)
+	if err != nil {
+		log.Errorf("Unable to subscribe to incoming ETH1.0 chain headers: %v", err)
+		w.runError = err
+		return
+	}
+	header, err = w.blockFetcher.HeaderByNumber(w.ctx, nil)
+	if err != nil {
+		log.Errorf("Unable to retrieve latest ETH1.0 chain header: %v", err)
+		w.runError = err
+		return
+	}
+	w.blockHeight = header.Number
+	w.blockHash = header.Hash()
+}
```

Comment:
Clear out `w.runError` if you are able to successfully resubscribe. Having a w.runError is an unhealthy condition and the process may be terminated if it cannot recover in a reasonable amount of time.

Reply:
done